### PR TITLE
Update Finch backend

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi,

This small PR updates Finch backend with what we currently have in the latest `finch-tensor` release.

### CODE REVIEW

File: sparse/finch_backend/__init__.py [hunk 1]
```diff
@@ -3,11 +3,93 @@
...
+    Element,
+    Pattern,
+    RepeatRLE,
+    SparseByteMap,
+    SparseCOO,
+    SparseHash,
+    SparseList,
+    SparseVBL,
+    Storage,
+    Tensor,
```
Comment (hameerabbasi) [inline sparse/finch_backend/__init__.py:18]: I would keep the API as Finch-agnostic as possible.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]

File: sparse/finch_backend/__init__.py (hunk 1 above)
Comment (mtsokol) [reply sparse/finch_backend/__init__.py:18]: But to define e.g. `Storage(Dense(...))` we would need to `import finch` separately. I think for now we can keep it in the `finch_backend` namespace, WDYT?

Comment (mtsokol) [reply sparse/finch_backend/__init__.py]: @hameerabbasi Done! I removed Finch-specific objects from the import list.

**Review by hameerabbasi — APPROVED** [review]
LGTM, thanks @mtsokol

**Pull request merged**